samples/snippets/custom_infotype.py

```diff
@@ -17,15 +17,434 @@
 custom infoType detectors to refine scan results.
 """
 
+# [START dlp_inspect_string_with_exclusion_dict]
+def inspect_string_with_exclusion_dict(
+    project, content_string, exclusion_list=["example@example.com"]
+):
+    """Inspects the provided text, avoiding matches specified in the exclusion list
+
+    Uses the Data Loss Prevention API to omit matches on EMAIL_ADDRESS if they are
+    in the specified exclusion list.
+
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        content_string: The string to inspect.
+        exclusion_list: The list of strings to ignore matches on
+
+    Returns:
+        None; the response from the API is printed to the terminal.
+    """
+
+    # Import the client library.
+    import google.cloud.dlp
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Construct a list of infoTypes for DLP to locate in `content_string`. See
+    # https://cloud.google.com/dlp/docs/concepts-infotypes for more information
+    # about supported infoTypes.
+    info_types_to_locate = [{"name": "EMAIL_ADDRESS"}]
+
+    # Construct a rule set that will only match on EMAIL_ADDRESS
+    # if the match text is not in the exclusion list.
+    rule_set = [
+        {
+            "info_types": info_types_to_locate,
+            "rules": [
+                {
+                    "exclusion_rule": {
+                        "dictionary": {
+                            "word_list": {
+                                "words": exclusion_list
+                            },
+                        },
+                        "matching_type": google.cloud.dlp_v2.MatchingType.MATCHING_TYPE_FULL_MATCH,
+                    }
+                }
+            ],
+        }
+    ]
+
+    # Construct the configuration dictionary
+    inspect_config = {
+        "info_types": info_types_to_locate,
+        "rule_set": rule_set,
+        "include_quote": True,
+    }
+
+    # Construct the `item`.
+    item = {"value": content_string}
+
+    # Convert the project id into a full resource id.
+    parent = f"projects/{project}"
+
+    # Call the API.
+    response = dlp.inspect_content(
+        request={"parent": parent, "inspect_config": inspect_config, "item": item}
+    )
+
+    # Print out the results.
+    if response.result.findings:
+        for finding in response.result.findings:
+            try:
+                if finding.quote:
+                    print(f"Quote: {finding.quote}")
+            except AttributeError:
+                pass
```

Comment:
Please don't swallow errors like this. Either log the error or let it bubble up.

Reply:
It's some boilerplate copied from the existing samples. Now that I'm looking, I agree that it's pretty icky. I'll clean it up throughout.

Reply:
Done.